src/navigation/tab.tsx

```diff
@@ -54,6 +59,23 @@ function HomeNavigator() {
 }
 
 export default function TabNavigator() {
+    const {isAuthenticated} = useSelector(
+        (state: RootStateOrAny) => state.auth,
+    );
+    useSelector(state => state.locale); // To re-render on any change in locale state
```

Comment:
This might not be needed here, since the re-render already happens from a higher component.